src/protocols/screencopy.rs

```diff
@@ -42,7 +85,28 @@ impl ScreencopyManagerState {
         };
         display.create_global::<D, ZwlrScreencopyManagerV1, _>(VERSION, global_data);
 
-        Self
+        Self {
+            queues: HashMap::new(),
+        }
+    }
+
+    pub fn bind(&mut self, manager: &ZwlrScreencopyManagerV1) {
+        self.queues.insert(manager.clone(), ScreencopyQueue::new());
+    }
+
+    pub fn destroy(&mut self, manager: &ZwlrScreencopyManagerV1) {
+        self.queues.remove(manager);
```

Comment:
Well I might be wrong but I thought removing the queue does not destroy the frame objects? This will fail all the frame requests of course but that is expected.

Reply:
You are correct, but as far as I read the protocol, this is not what should happen. I.e. if the client creates a manager, then calls capture_output(), then destroys a manager, the frame can still get rendered, since "All objects created by the manager will still remain valid, until their appropriate destroy request has been called."

Reply:
Mhmmm I am not sure. If the client wants to `copy_with_damage` which will put it in the queue, but the manger instance has already been destroyed, hence it would not be possible to compute the damage.

Reply:
Why not? The manager is not alive, but the Rust object still exists. You just can't send any events to it.

Reply:
I think you are right. I will implement it as described in your first comment. I blame the underspecified documentation :p

Reply:
This is a bit tricky since we have to add it to every site that might remove a screencopy from a queue, and can be  error prone to keep track of them all. :(

Reply:
You could make a wrapper method around queue.pop() that also checks if the manager is still alive?

Reply:
Yeah well the problem is that ScreencopyQueue of course can't remove itself from the hashmap it is in. The best way I can think of is let `ScreencopyManagerState::get_queue_mut` take a closure, run it and then check for the emptiness at the end.